File: apps/mobile/test/generated_client_test.dart
import 'package:argent_api_client/argent_api_client.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('generated Dart client preserves liveness fields', () {
    final health = ServiceHealth(
      (builder) => builder
        ..service = 'argent-api'
        ..state = ServiceHealthStateEnum.ok
        ..version = 'contract-test',
    );

    expect(health.service, 'argent-api');
    expect(health.state, ServiceHealthStateEnum.ok);
    expect(health.version, 'contract-test');
  });
}

File: packages/contracts/test/generated-client.test.ts
import {
  Configuration,
  type FetchAPI,
  ServiceHealthFromJSON,
  ServiceHealthStateEnum,
  SystemApi,
} from "@argent/api-client";
import { describe, expect, it } from "vitest";

describe("generated TypeScript client", () => {
  it("maps a liveness response without losing enum semantics", () => {
    expect(
      ServiceHealthFromJSON({
        service: "argent-api",
        state: "ok",
        version: "contract-test",
      }),
    ).toEqual({
      service: "argent-api",
      state: ServiceHealthStateEnum.Ok,
      version: "contract-test",
    });
  });

  it("exposes the versioned system operation", () => {
    expect(typeof new SystemApi().getLiveness).toBe("function");
  });

  it("calls the documented route and decodes its response", async () => {
    const requests: string[] = [];
    const fetchApi: FetchAPI = async (input) => {
      requests.push(input.toString());

      return new Response(
        JSON.stringify({
          service: "argent-api",
          state: "ok",
          version: "contract-test",
        }),
        {
          headers: { "content-type": "application/json" },
          status: 200,
        },
      );
    };
    const client = new SystemApi(
      new Configuration({
        basePath: "https://api.argent.test",
        fetchApi,
      }),
    );

    await expect(client.getLiveness()).resolves.toEqual({
      service: "argent-api",
      state: ServiceHealthStateEnum.Ok,
      version: "contract-test",
    });
    expect(requests).toEqual(["https://api.argent.test/health/live"]);
  });
});
